common/math/integer.go

```diff
@@ -96,3 +96,11 @@ func SafeMul(x, y uint64) (uint64, bool) {
 	hi, lo := bits.Mul64(x, y)
 	return lo, hi != 0
 }
+
+// GetAbsoluteDifference is a utility method that given 2 int64, it returns the absolute value of their difference in uint64 format.
+func GetAbsoluteDifference(x, y uint64) uint64 {
+	if x > y {
```

Comment:
why not something like `math.Abs(x - y)`?

Reply:
math.Abs takes floating point and return floating points, and uint64 to float64 casting are dangerous.